executor/table_reader.go

```diff
@@ -275,16 +262,16 @@ func (e *TableReaderExecutor) Next(ctx context.Context, req *chunk.Chunk) error
 
 // Close implements the Executor Close interface.
 func (e *TableReaderExecutor) Close() error {
-	var err error
 	if e.resultHandler != nil {
-		err = e.resultHandler.Close()
+		if err := e.resultHandler.Close(); err != nil {
+			return err
+		}
 	}
 	e.kvRanges = e.kvRanges[:0]
 	if e.dummy {
 		return nil
```

Comment:
Looks like the logic here is different from before.
Previously, errors from `e.resultHandler.Close()` won't make it return directly.

Reply:
Good catch! Updated.